lib/dependencies.sh

```diff
@@ -59,12 +59,28 @@ log_build_scripts() {
   fi
 }
 
+yarn_supports_frozen_lockfile() {
+  local yarn_version="$(yarn --version)"
+  # Yarn versions lower than 0.19 will crash if passed --frozen-lockfile
+  if [[ "$yarn_version" =~ ^0\.(16|17|18).*$ ]]; then
+    mcount "yarn.doesnt-support-frozen-lockfile"
+    echo false
+  else
+    echo true
+  fi
+}
+
 yarn_node_modules() {
   local build_dir=${1:-}
+  local supports_frozen_lockfile=$(yarn_supports_frozen_lockfile)
 
   echo "Installing node modules (yarn.lock)"
   cd "$build_dir"
-  yarn install --pure-lockfile --ignore-engines 2>&1
+  if [ "$supports_frozen_lockfile" = true ]; then
```

Comment:
`if yarn_supports_frozen_lockfile; then` (and, generally, always use `[[` with `if` if you require statements inside)